File: my-folder/0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int minLength = nums.length+1;
        int currSum = 0;
        while (right < nums.length) {
            currSum += nums[right];
            while (currSum >= target) {
                minLength = Math.min(minLength, right - left + 1);
                currSum -= nums[left];
                left++;
            }
            right++;
        }

        if (minLength == nums.length+1) return 0;
        return minLength;
    }
}

File: my-folder/0209-minimum-size-subarray-sum/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = right = 0
        curr_sum = 0
        res = len(nums) + 1
        while right < len(nums):
            curr_sum += nums[right]
            while curr_sum >= target:
                res = min(res, right - left + 1)
                curr_sum -= nums[left]
                left += 1
            right += 1

        if res == len(nums) + 1: return 0
        return res
